### PR TITLE
Switch kvm-operator to ignition

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/2816

### CODE REVIEW

File: service/controller/v18/cloudconfig/cloudconfigtest/cloudconfigtest.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package cloudconfigtest
 import (
 	"github.com/giantswarm/micrologger/microloggertest"
 
+	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v_4_0_0"
```
Comment (xh3b4sd) [inline service/controller/v18/cloudconfig/cloudconfigtest/cloudconfigtest.go]:
```suggestion
	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v_4_0_0"

```

File: service/controller/v18/cloudconfig/cloudconfigtest/cloudconfigtest.go [hunk 2]
```diff
@@ -3,6 +3,7 @@ package cloudconfigtest
 import (
 	"github.com/giantswarm/micrologger/microloggertest"
```
Comment (xh3b4sd) [inline service/controller/v18/cloudconfig/cloudconfigtest/cloudconfigtest.go]:
```suggestion
```

File: service/controller/v18/cloudconfig/master_template.go [hunk 3]
```diff
@@ -15,10 +17,11 @@ func (c *CloudConfig) NewMasterTemplate(customObject v1alpha1.KVMConfig, certs c
 
 	var params k8scloudconfig.Params
 	{
+		params = k8scloudconfig.DefaultParams()
+
 		params.APIServerEncryptionKey = string(randomKeys.APIServerEncryptionKey)
+		params.BaseDomain = strings.TrimPrefix(customObject.Spec.Cluster.Kubernetes.API.Domain, "api.")
```
Comment (xh3b4sd) [inline service/controller/v18/cloudconfig/master_template.go]: Let's please have a key package method for the magic string mutations.

File: service/controller/v18/key/key.go [hunk 4]
```diff
@@ -29,7 +29,7 @@ const (
 	// ProbeHost host for liveness probe.
 	ProbeHost = "127.0.0.1"
 	// InitialDelaySeconds is InitialDelaySeconds param in liveness probe config
```
Comment (xh3b4sd) [inline service/controller/v18/key/key.go]:
```suggestion
	// InitialDelaySeconds is InitialDelaySeconds param in liveness probe config.
```

File: service/controller/v18/key/key.go [hunk 5]
```diff
@@ -29,7 +29,7 @@ const (
...
-	InitialDelaySeconds = 240
+	InitialDelaySeconds = 360
 	// TimeoutSeconds is TimeoutSeconds param in liveness probe config
```
Comment (xh3b4sd) [inline service/controller/v18/key/key.go]:
```suggestion
	// TimeoutSeconds is TimeoutSeconds param in liveness probe config.
```

File: service/controller/v18/key/key.go [hunk 6]
```diff
@@ -29,7 +29,7 @@ const (
...
 	TimeoutSeconds = 5
 	// PeriodSeconds is PeriodSeconds param in liveness probe config
```
Comment (xh3b4sd) [inline service/controller/v18/key/key.go]:
```suggestion
	// PeriodSeconds is PeriodSeconds param in liveness probe config.
```

File: service/controller/v18/key/key.go [hunk 7]
```diff
@@ -29,7 +29,7 @@ const (
 	// ProbeHost host for liveness probe.
 	ProbeHost = "127.0.0.1"
 	// InitialDelaySeconds is InitialDelaySeconds param in liveness probe config
-	InitialDelaySeconds = 240
+	InitialDelaySeconds = 360
```
Comment (xh3b4sd) [inline service/controller/v18/key/key.go]: Why is that 6 minutes now?

File: service/controller/v18/key/key.go [hunk 8]
```diff
@@ -48,10 +48,10 @@ const (
 
 	FlannelEnvPathPrefix = "/run/flannel"
 	CoreosImageDir       = "/var/lib/coreos-kvm-images"
-	CoreosVersion        = "1855.5.0"
+	CoreosVersion        = "1911.5.0"
```
Comment (xh3b4sd) [inline service/controller/v18/key/key.go]: @giantswarm/sig-operator why do we define that here? Don't we do that in `k8scloudconfig`?

File: service/controller/v18/cloudconfig/worker_template.go [hunk 9]
```diff
@@ -3,8 +3,9 @@ package cloudconfig
 import (
 	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
 	"github.com/giantswarm/certs"
-	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v_3_7_3"
+	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v_4_0_0"
```
Comment (xh3b4sd) [inline service/controller/v18/cloudconfig/worker_template.go]: Do we plan to rename the repo at some point? What is the idea here? /cc @giantswarm/sig-operator

File: service/controller/v18/version_bundle.go [hunk 10]
```diff
@@ -12,6 +12,16 @@ func VersionBundle() versionbundle.Bundle {
 				Description: "Add k8s api and kubelet health check to kvm pod.",
 				Kind:        versionbundle.KindAdded,
 			},
+			{
+				Component:   "cloudconfig",
```
Comment (xh3b4sd) [inline service/controller/v18/version_bundle.go]: Should we rename the component as well at some point?

File: service/controller/v18/version_bundle.go [hunk 11]
```diff
@@ -36,6 +46,6 @@ func VersionBundle() versionbundle.Bundle {
 			},
 		},
 		Name:    "kvm-operator",
-		Version: "3.2.1",
+		Version: "3.2.0",
```
Comment (xh3b4sd) [inline service/controller/v18/version_bundle.go]: Why is that a downgrade? Should at least be a minor bump IMO.

Comment (corest) [reply service/controller/v18/version_bundle.go]: the previous version is `3.1.1`. I don't know why it got `3.2.1` and not `3.2.0`.

File: service/controller/v18/key/key.go (hunk 7 above)
Comment (xh3b4sd) [reply service/controller/v18/key/key.go]: Are there opportunities to speed this up in the future? SIG operator would like to pursue a main objective of fast transitioning times, which includes tenant cluster creation.

Comment (corest) [reply service/controller/v18/key/key.go]: I've spent a lot of time here and the only way we can speed up the process - burning all static configurations into  coreos image. That would require CI process for building pxe images, ami images for amazon and azure images.

File: service/controller/v18/cloudconfig/worker_template.go (hunk 9 above)
Comment (kopiczko) [reply service/controller/v18/cloudconfig/worker_template.go]: I think so. We can do that together with VOO when we won't have to use version packages.

File: service/controller/v18/key/key.go (hunk 8 above)
Comment (kopiczko) [reply service/controller/v18/key/key.go]: I have mixed feelings. E.g. in AWS we need to know the AMI. I wouldn't put that info into `k8scloudconfig`. Also releases arrive on different time on different providers and from time to time we need bugfixes specifically for a provider.

**Review by kopiczko — APPROVED** [review]
LGTM when tested.

**Pull request merged**